packages/flutter/lib/src/widgets/scroll_metrics.dart

```diff
@@ -74,12 +75,21 @@ abstract class ScrollMetrics {
   /// [minScrollExtent]. It can be infinity, if the scroll is unbounded.
   double get maxScrollExtent;
 
+  /// Whether [maxScrollExtent] property is available.
+  bool get hasMaxScrollExtent;
+
   /// The current scroll position, in logical pixels along the [axisDirection].
   double get pixels;
 
+  /// Whether [pixels] property is available.
+  bool get hasPixels;
+
   /// The extent of the viewport along the [axisDirection].
   double get viewportDimension;
 
+  /// Whether [viewportDimension] property is available.
+  bool get hasViewportDimension;
```

Comment:
(for all of these) "Whether _the_ [foo] property is available"

Reply:
Done